docs/source/theory.rst

```diff
@@ -4,5 +4,337 @@
 Theory
 ======
 
+--------------
+Bulk physics 
+--------------
 
-WIP
\ No newline at end of file
+H transport
+^^^^^^^^^^^
+
+The model developed by McNabb & Foster [:ref:`1 <McNabb and Foster>`] is used to model hydrogen transport in materials in FESTIM. The principle is to separate mobile hydrogen :math:`c_\mathrm{m}` and trapped hydrogen :math:`c_\mathrm{t}`. The diffusion of mobile particles is governed by Fick’s law of diffusion where the hydrogen flux is
+
+.. math::
+    :label: eq_difflux
+    
+    J = -D \nabla c_\mathrm{m}
+
+where :math:`D=D(T)` is the diffusivity. Each trap :math:`i` is associated with a 
+trapping and a detrapping rate :math:`k_i` and :math:`p_i`, respectively, as well 
+as a trap density :math:`n_i`.
```

Comment:
@RemDelaporteMathurin, possibly, we can add here something about the Arrhenius form, namely: 

The diffusivity is usually assumed to follow the Arrhenius law of temperature **[Reference?]:** $D(T)=D_0e^{-\frac{E_d}{kT}}$, where $D_0$ is the diffusivity prefactor, $E_d$ - diffusion activation energy, $k$ - the Boltzmann constant, $T$ - temperature.

Reply:
we could have something more general since it applies not only to the diffusivity but also to trapping rates, solubilities....

Reply:
Therefore, a separate subsection somewhere? The first mention is in the **Heat transfer** section.

Reply:
Possibly, [this paper](https://doi.org/10.1021/ed061p494) is appropriate for a reference. Original works of Arrhenius are in deutish as I see

Reply:
Yes maybe a separate subsection, no need for a reference

Reply:
@RemDelaporteMathurin, I made a try